apps/beeai-server/src/beeai_server/domain/model.py

```diff
@@ -213,6 +240,19 @@ class Provider(BaseModel):
     manifest: ProviderManifest
     id: ID
     registry: GithubUrl | None = None
+    env: dict[str, str] | None = None
+
+    @asynccontextmanager
+    async def mcp_client(self) -> McpClient:
+        async with self.manifest.mcp_client(env=self.env) as client:
+            yield client
+
+    @model_validator(mode="after")
+    def level_uvicorn_validator(self):
+        required_vars = set(var.name for var in self.manifest.env if var.required)
+        if missing_vars := required_vars - (self.env or {}).keys():
```

Comment:
Should we implicitly pass the env if it's not found in the provider config, but it's found in actual `os.environ`?